File: code-coaches/medium/divisible/Divisible.java
import java.util.Scanner;

public class Divisible {

    public static boolean isDivisibleByAll(int num, String[] dividersString) {
        for (String dividerString : dividersString) {
            int divider = Integer.parseInt(dividerString);
            if (num % divider != 0) {
                return false;
            }
        }

        return true;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int num = scanner.nextInt();
        scanner.nextLine();
        String[] dividersString = scanner.nextLine().split(" ");

        if (isDivisibleByAll(num, dividersString)) {
            System.out.println("divisible by all");
        } else {
            System.out.println("not divisible by all");
        }

        scanner.close();
    }

}

File: code-coaches/medium/divisible/divisible.py
def is_divisible_by_all(num: int, dividers: list[int]) -> bool:
    for divider in dividers:
        if num % divider != 0:
            return False

    return True


def main():
    num = int(input())
    dividers = list(map(int, input().split()))

    if is_divisible_by_all(num, dividers):
        print("divisible by all")
    else:
        print("not divisible by all")


if __name__ == "__main__":
    main()
